courseography.cabal

```diff
@@ -34,25 +33,63 @@ test-suite Tests
     Database.Tables,
     DynamicGraphs.GraphNodeUtils,
     Svg.Builder,
-    Tests.Controllers.CourseControllerTests,
-    Tests.RequirementTests.ModifierTests,
-    Tests.RequirementTests.PostParserTests,
-    Tests.RequirementTests.PreProcessingTests,
-    Tests.RequirementTests.ReqParserTests,
-    Tests.RequirementTests.RequirementTests,
-    Tests.TestHelpers,
     Util.Happstack,
+    WebParsing.ArtSciParser,
     WebParsing.ParsecCombinators,
     WebParsing.PostParser,
     WebParsing.ReqParser,
-    WebParsing.ArtSciParser
+  build-depends:
+    aeson,
+    base >=4.14,
+    bytestring,
+    Cabal >= 1.16.0,
+    containers,
+    directory,
+    happstack-server,
+    hslogger,
+    http-conduit,
+    http-types,
+    MissingH,
+    monad-logger,
+    mtl,
+    parsec,
+    persistent,
+    persistent-sqlite,
+    resourcet,
+    split,
+    system-filepath,
+    tagsoup,
+    text,
+    time,
+    transformers,
+    unordered-containers,
+    yaml
+  default-extensions:
+    OverloadedStrings,
+    ScopedTypeVariables
+  default-language: Haskell2010
+  ghc-options: -Wall -fwarn-tabs
+  hs-source-dirs: app
+
+
+test-suite Tests
+  main-is: Main.hs
+  other-modules:
+    Controllers.CourseControllerTests,
+    RequirementTests.ModifierTests,
+    RequirementTests.PostParserTests,
+    RequirementTests.PreProcessingTests,
+    RequirementTests.ReqParserTests,
+    RequirementTests.RequirementTests,
+    TestHelpers
   Type: exitcode-stdio-1.0
   build-depends:
     aeson,
     base >=4.14,
     bytestring,
     Cabal >= 1.16.0,
     containers,
+    courseography,
```

Comment:
So it turns out not all of them can be removed. I believe the ones that can be removed are the ones not explicitly used within the testing library. So to be specific, I could remove yaml because it wasn't explicitly being imported in the test suite. I have gone ahead and removed all such, but the remaining ones will have to be kept I think.